src/collectors/mesos/test/testmesos.py

```diff
@@ -25,25 +25,97 @@ def setUp(self):
     def test_import(self):
         self.assertTrue(MesosCollector)
 
+    def test_import(self):
+        self.assertTrue(self.collector.config['path'], 'mesos')
+
     @patch.object(Collector, 'publish')
-    def test_should_work_with_real_data(self, publish_mock):
-        def se(url):
-            if url == 'http://localhost:5050/metrics/snapshot':
-                return self.getFixture('metrics')
+    def test_should_work_for_master_with_real_data(self, publish_mock):
+        returns = self.getFixture('master_metrics_snapshot.json')
+        urlopen_mock = patch('urllib2.urlopen', Mock(
+            side_effect=lambda *args: returns))
 
-        patch_urlopen = patch('urllib2.urlopen', Mock(side_effect=se))
+        urlopen_mock.start()
+        self.collector.collect()
+        urlopen_mock.stop()
 
-        patch_urlopen.start()
+        # check how many fixtures were consumed
+        self.assertEqual(urlopen_mock.new.call_count, 1)
+
+        metrics = {
+            'master.elected': (1, 0),
+            "system.mem_free_bytes": (5663678464.1, 0),
+            "registrar.state_store_ms.p9999": (17.8412544, 6)
+        }
+
+        self.setDocExample(collector=self.collector.__class__.__name__,
```

Comment:
`self.setDocExample` should only be called once (already called in `test_should_work_for_slave_with_real_data`).  Remove this one or the other one